### PR TITLE
APEXMALHAR-2242 Added user documentation for 0.9 version of Kafka Input Operator

### PR DESCRIPTION
**Pull request by chaithu14**

### CODE REVIEW

File: docs/operators/kafkaInputOperator.md
Comment (amberarrow) [inline docs/operators/kafkaInputOperator.md:27]: Since both have the same name, add the the package name in brackets for both so users can distinguish them.

File: docs/operators/kafkaInputOperator.md
Comment (amberarrow) [inline docs/operators/kafkaInputOperator.md]: Rewrite both sentences something like this:
This version uses the new 0.9 version of the Consumer API; differences between the two API versions are described here <provide a suitable link>.
This operator is fault-tolerant, scalable, and has multi-cluster and multi-topic support.

File: docs/operators/kafkaInputOperator.md
Comment (amberarrow) [inline docs/operators/kafkaInputOperator.md]: Rewrite:
This operator requires version 0.9.0 or later of the Kafka Consumer API.

**Comment by chaithu14** [comment]
Incorporated the review comments and updated the PR.

File: docs/operators/kafkaInputOperator.md
Comment (tweise) [reply docs/operators/kafkaInputOperator.md:27]: It's not the 0.8 version of the operator but the operator for Kafka 0.8.x

File: docs/operators/kafkaInputOperator.md
Comment (tweise) [inline docs/operators/kafkaInputOperator.md:301]: I think this document should explain what the ready to use operators are. AbstractXXX is for implementing custom operator.

File: docs/operators/kafkaInputOperator.md
Comment (tweise) [inline docs/operators/kafkaInputOperator.md:386]: Instead of putting all the code here, the example will probably be more useful for developers.

**Review by tweise — CHANGES_REQUESTED** [review]

File: docs/operators/kafkaInputOperator.md
Comment (chaithu14) [reply docs/operators/kafkaInputOperator.md:27]: Done.

File: docs/operators/kafkaInputOperator.md
Comment (chaithu14) [reply docs/operators/kafkaInputOperator.md:301]: Concrete operators are also covered in next section.

File: docs/operators/kafkaInputOperator.md
Comment (chaithu14) [reply docs/operators/kafkaInputOperator.md:386]: Link to example is provided at the end of doc.

**Comment by tweise** [comment]
I think that currently the documentation repeats a lot of javadoc, instead it should explain how common tasks are solved. See 

https://issues.apache.org/jira/browse/APEXMALHAR-2287
https://issues.apache.org/jira/browse/APEXMALHAR-2288

Note what 2287 is asking for, it's based on the questions on the users list and not really covered here.

@amberarrow thoughts?

**Comment by amberarrow** [comment]
Yes, some narrative style text that discusses how to use this operator in a couple of common use cases would be helpful. Some of the descriptions of the configurable parameters have useful data.

My initial thought is a "How to use" section that starts by discussing the 0.8 and 0.9 APIs, then input versus output, then the notion of topics and partitions and how they map to operator partitions, then performance related issues, exactly once, and closing with troubleshooting common problems. Not cast in stone but something along those lines.

**Comment by chaithu14** [comment]
Thanks @tweise and @amberarrow for reviewing.
Please consider this as initial version of doc and merge this PR. @tweise created JIRA's for each of the versions. I will work on those new JIRA's.

**Comment by tweise** [comment]
There are a few basic changes that haven't been made. But instead of more ping pong I will submit these for inclusion into the PR and then it will need to be rebased.

**Comment by tweise** [comment]
@chaithu14 sent you PR for the changes. LMK when this PR is rebased.

**Comment by chaithu14** [comment]
@tweise Re-based the changes and updated the PR.

File: docs/operators/kafkaInputOperator.md
Comment (amberarrow) [inline docs/operators/kafkaInputOperator.md]: consume => consumes

File: docs/operators/kafkaInputOperator.md
Comment (amberarrow) [inline docs/operators/kafkaInputOperator.md]: consume => consumes
"If there is not committed offset," => If a committed offset is unavailable,

File: docs/operators/kafkaInputOperator.md
Comment (amberarrow) [inline docs/operators/kafkaInputOperator.md]: Specifies ... => If set to a value other than the default, such as `FSWindowDataManager`, specifies ....

File: docs/operators/kafkaInputOperator.md
Comment (amberarrow) [inline docs/operators/kafkaInputOperator.md]: Please surround code fragments with backquotes.

File: docs/operators/kafkaInputOperator.md
Comment (amberarrow) [inline docs/operators/kafkaInputOperator.md]: getTuple() needs backquotes around it.

File: docs/operators/kafkaInputOperator.md
Comment (amberarrow) [inline docs/operators/kafkaInputOperator.md]: backquotes needed

File: docs/operators/kafkaInputOperator.md
Comment (amberarrow) [inline docs/operators/kafkaInputOperator.md]: This link (and others below) should be formatted as markdown links.

**Comment by tweise** [comment]
@chaithu14 there was no need to squash the extra commit since it wasn't yours. Can you give me write access to your fork so I can make the remaining edits, update the PR and merge it.

**Comment by tweise** [comment]
@amberarrow addressed comments, please merge.

File: docs/operators/kafkaInputOperator.md
Comment (amberarrow) [inline docs/operators/kafkaInputOperator.md]: This link doesn't show as a clickable link in the rendered page.

File: docs/operators/kafkaInputOperator.md
Comment (amberarrow) [inline docs/operators/kafkaInputOperator.md:313]: This should be formatted as an HTML table to be consistent with the others above.

**Comment by amberarrow** [comment]
Made a couple of additional comments; I'll merge after those are addressed.

File: docs/operators/kafkaInputOperator.md
Comment (tweise) [reply docs/operators/kafkaInputOperator.md:313]: This wasn't brought up earlier and can be addressed separately. Should also consider which format is more suitable.

**Comment by tweise** [comment]
Fixed link, please merge.

**Pull request merged**
